### PR TITLE
Add edr cube

### PR DESCRIPTION
**Pull request by ernstdevreede**
This PR adds the cube call to ADAGUC EDR for gridded data.

### CODE REVIEW

File: python/python_fastapi_server/routers/netcdf_to_covjson.py [hunk 1]
```diff
@@ -198,7 +239,7 @@ def netcdf_to_covjson(netcdfdataset) -> Coverage:
     import requests
 
     SERVICE = "https://geoservices.knmi.nl/adagucserver?"
-    SERVICE= "http://localhost:8080/adagucserver?"
+    SERVICE = "http://localhost:8080/adagucserver?"
```
Comment (maartenplieger) [inline python/python_fastapi_server/routers/netcdf_to_covjson.py]: Should this stay? Is this code touched?

File: python/python_fastapi_server/routers/netcdf_to_covjson.py [hunk 2]
```diff
@@ -264,19 +305,23 @@ def netcdf_to_covjson(netcdfdataset) -> Coverage:
...
 
+    QUERYSTRING = "dataset=netcdf_5d&service=wcs&version=1.1.1&request=getcoverage&format=NetCDF4&crs=EPSG:4326&coverage=data&bbox=4.603,51.27,6.274,52.624&time=2017-01-01T00:00Z/2017-01-01T00:30Z&elevation=1000/9000"
+
     WCSGETCOVERAGEURL = SERVICE + QUERYSTRING
 
     # Get a NetCDF file as dataset
     response = requests.get(WCSGETCOVERAGEURL, timeout=60)
 
     print(response.status_code)
+    with open("/tmp/klad.nc", "wb") as f:
```
Comment (maartenplieger) [inline python/python_fastapi_server/routers/netcdf_to_covjson.py]: Probably remove?

Comment (maartenplieger) [reply python/python_fastapi_server/routers/netcdf_to_covjson.py]: Or make a bit nicer. I see it is meant for testing purposes.

File: python/python_fastapi_server/routers/edr_utils.py [hunk 3]
```diff
@@ -0,0 +1,808 @@
...
+        ref_time,
+        edr_collectioninfo["time_interval"],
+        time_interval,
+        interval,
+        times,
+    )
+    return interval, times
+
+
+SYMBOL_TYPE_URL = "http://www.opengis.net/def/uom/UCUM"
```
Comment (maartenplieger) [inline python/python_fastapi_server/routers/edr_utils.py]: Please move to top

File: python/python_fastapi_server/routers/edr_covjson.py [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+from covjson_pydantic.reference_system import (
+    ReferenceSystem,
+    ReferenceSystemConnectionObject,
+)
+from covjson_pydantic.unit import Symbol as CovJsonSymbol
+from covjson_pydantic.unit import Unit as CovJsonUnit
+from pydantic import AwareDatetime
+
+from .edr_exception import EdrException
+from .edr_utils import get_edr_collections, get_param_metadata
```
Comment (maartenplieger) [inline python/python_fastapi_server/routers/edr_covjson.py]: get_edr_collections is not used

File: python/python_fastapi_server/routers/edr_locations.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+This code uses Adaguc's OGC WMS and WCS endpoints to convert into an EDR service.
+
+Author: Ernst de Vreede, 2023-11-23
+
+KNMI
+"""
+
+from fastapi import APIRouter
+
+import logging
```
Comment (maartenplieger) [inline python/python_fastapi_server/routers/edr_locations.py:11]: Please move above `from fastapi import APIRouter`

File: python/python_fastapi_server/routers/edr_utils.py [hunk 6]
```diff
@@ -0,0 +1,808 @@
...
+            data_queries=data_queries,
+            parameter_names=parameter_names,
+            crs=crs,
+            output_formats=output_formats,
+        )
+
+    return collection, ttl
+
+
+def parse_period_string(period: str):
```
Comment (maartenplieger) [inline python/python_fastapi_server/routers/edr_utils.py:456]: Can you provide a small descriptive docstring for this function?

File: python/python_fastapi_server/routers/edr_utils.py [hunk 7]
```diff
@@ -0,0 +1,808 @@
...
+    elif step_type == "h":
+        delta = relativedelta(hours=step)
+    elif step_type == "m":
+        delta = relativedelta(minutes=step)
+    elif step_type == "s":
+        delta = relativedelta(seconds=step)
+    return delta
+
+
+def parse_interval_string(time_interval: str, ref_time: datetime):
```
Comment (maartenplieger) [inline python/python_fastapi_server/routers/edr_utils.py:486]: Can you provide a small descriptive docstring for this function?

File: doc/configuration/EDRConfiguration/EDR.md [hunk 8]
```diff
@@ -8,17 +8,32 @@ The OGC API EDR service can be configured in a dataset configuration by adding t
...
+        <EdrCollection name="....">
+          <EdrParameter name="..."/>
         </EdrCollection>
     </OgcApiEdr>
 ```
 
 EdrCollection:
 - name: The name of the EDR collection
-- vertical_dimension?: For example pressure_level
+- vertical_name?: For example pressure_level, the name of the vertical dimension in the WMS layer configuration
```
Comment (maartenplieger) [inline doc/configuration/EDRConfiguration/EDR.md:26]: Did you check `Configure_EDR_service.md` as well?

File: doc/configuration/EDRConfiguration/EDR.md [hunk 9]
```diff
@@ -8,17 +8,32 @@ The OGC API EDR service can be configured in a dataset configuration by adding t
 
 ```xml
```
Comment (maartenplieger) [inline doc/configuration/EDRConfiguration/EDR.md:13]: Can you add above that currently position and cube are supported?

File: python/python_fastapi_server/routers/netcdf_to_covjson.py (hunk 1 above)
Comment (ernstdevreede) [reply python/python_fastapi_server/routers/netcdf_to_covjson.py]: Removed this not always working testcode

File: python/python_fastapi_server/routers/netcdf_to_covjson.py (hunk 2 above)
Comment (ernstdevreede) [reply python/python_fastapi_server/routers/netcdf_to_covjson.py]: removed

File: python/python_fastapi_server/routers/edr_utils.py (hunk 3 above)
Comment (ernstdevreede) [reply python/python_fastapi_server/routers/edr_utils.py]: Moved

File: python/python_fastapi_server/routers/edr_covjson.py (hunk 4 above)
Comment (ernstdevreede) [reply python/python_fastapi_server/routers/edr_covjson.py]: Removed

File: python/python_fastapi_server/routers/edr_locations.py (hunk 5 above)
Comment (ernstdevreede) [reply python/python_fastapi_server/routers/edr_locations.py:11]: Done

File: doc/configuration/EDRConfiguration/EDR.md (hunk 9 above)
Comment (ernstdevreede) [reply doc/configuration/EDRConfiguration/EDR.md:13]: Done

File: python/python_fastapi_server/routers/edr_utils.py (hunk 7 above)
Comment (ernstdevreede) [reply python/python_fastapi_server/routers/edr_utils.py:486]: Done

File: python/python_fastapi_server/routers/edr_utils.py (hunk 6 above)
Comment (ernstdevreede) [reply python/python_fastapi_server/routers/edr_utils.py:456]: Done

File: doc/configuration/EDRConfiguration/EDR.md (hunk 8 above)
Comment (ernstdevreede) [reply doc/configuration/EDRConfiguration/EDR.md:26]: Yes, added a few lines.

**Review by maartenplieger — APPROVED** [review]

**Pull request merged**